upstairs/src/lib.rs

```diff
@@ -1513,12 +1683,37 @@ struct DownstairsIO {
      * or if by not putting a downstars in the hash, if that is valid.
      */
     state: HashMap<u8, IOState>,
+    /*
+     * Has this been acked to the guest yet?
+     */
+    ack_status: AckStatus,
     /*
      * If the operation is a Read, this holds the resulting buffer
      */
     data: Option<Bytes>,
 }
 
+impl DownstairsIO {
+    fn state_count(&self) -> WorkCounts {
+        let mut wc: WorkCounts = Default::default();
+
+        for state in self.state.values() {
+            match state {
+                IOState::New | IOState::InProgress => wc.active += 1,
+                IOState::Error(_) => wc.error += 1,
```

Comment:
Nit, these three could all look like error, set the wc. on the same line.

Reply:
Done in e88800c